src/components/o-testimonials/js/testimonials.js

```diff
@@ -0,0 +1,163 @@
+    }
+    this.showSlide(0);
+    this.on();
+    this.initSwipe();
+  }
+
+  on() {
+    this.off();
+    this.controlLeftClicked = on(this.controlLeft, EVENTS.CLICK, this.handleControlLeftClicked);
+    this.controlRightClicked = on(this.controlRight, EVENTS.CLICK, this.handleControlRightClicked);
+  }
+
+  off() {
+    if (this.controlLeftClicked) {
+      this.controlLeftClicked();
+    }
+    if (this.controlRightClicked) {
+      this.controlRightClicked();
+    }
+  }
+
+  initSwipe() {
+    const swiper = new Swipe(this.slider);
+    swiper.onLeft(() => {
+      this.autoRotateEnabled = false;
+      this.showSlide(+1);
+    });
+    swiper.onRight(() => {
+      this.autoRotateEnabled = false;
+      this.showSlide(-1);
+    });
+    swiper.run();
+  }
+
+  handleControlLeftClicked = () => {
+    this.autoRotateEnabled = false;
+    this.showSlide(-1);
+  };
+
+  handleControlRightClicked = () => {
+    this.autoRotateEnabled = false;
+    this.showSlide(+1);
+  };
+
+  hideAllSlides() {
+    for (let i = 0, { length } = this.slides; i < length; i++) {
+      this.slides[i].style.display = 'none';
+    }
+  }
+
+  hideControls() {
+    this.controlRight.style.display = 'none';
+    this.controlLeft.style.display = 'none';
+  }
+
+  showSlide(positionDifference) {
+    this.hideAllSlides();
+    if (this.slideIndex + positionDifference < 0) {
+      this.slideIndex = this.slides.length - 1;
+    } else if (this.slideIndex + positionDifference >= this.slides.length) {
+      this.slideIndex = 0;
+    } else {
+      this.slideIndex = this.slideIndex + positionDifference;
+    }
+    this.slides[this.slideIndex].classList.remove('o-testimonials__item_animation_left');
+    this.slides[this.slideIndex].classList.remove('o-testimonials__item_animation_right');
+    if (positionDifference < 0) {
+      this.slides[this.slideIndex].classList.add('o-testimonials__item_animation_left');
+    } else {
+      this.slides[this.slideIndex].classList.add('o-testimonials__item_animation_right');
+    }
+    this.slides[this.slideIndex].style.display = 'block';
+    this.autoRotate();
+  }
+
+  autoRotate() {
+    // auto rotate until disabled
+    if (this.autoRotateEnabled === true || this.autoRotateEnabled === 'true') {
+      setTimeout(
+        () => {
+          if (this.autoRotateEnabled === true || this.autoRotateEnabled === 'true') {
+            // if disabled meanwhile
+            this.showSlide(+1);
+          }
+        },
+        this.autoRotateTimeInMiliseconds,
+      );
+    }
+  }
+
+  destroy() {
+    this.off();
+
+    if (this.controlLeft) {
+      delete this.controlLeft;
+    }
+
+    if (this.controlRight) {
+      delete this.controlRight;
+    }
+
+    if (this.wcNode) {
+      delete this.wcNode;
+    }
+
+    if (this.options) {
+      delete this.options;
+    }
+
+    if (this.slides) {
+      delete this.slides;
+    }
+
+    if (this.slider) {
+      delete this.slider;
+    }
+
+    delete this.handleClick;
```

Comment:
Why is `handleClick` deleted, I can't find it being defined here 🤔

Reply:
I will remove this line. I forgot to delete it from the initial implementation.